SparkPost/2web2ui
# FE-1160 - [Domains UX Redesign] Add a Domain page

[FE-1160](https://sparkpost.atlassian.net/browse/FE-1160)

### What Changed
- Adds new create domains page that renders at `/domains/create`
- Updates all Matchbox component wrappers for form controls to leverage `React.forwardRef` to make them more easily used with `react-hook-form`
- Updates Matchbox dependencies to version `4.2.1`
- Adds supporting Cypress tests

### How To Test

1. On an account with the `"allow_domains_v2"` flag, go to `/domains/create` and create a sending domain and a tracking domain
1. Verify you see a success message and are re-directed to the verification page
1. Verify network requests have relevant data when made
1. Verify sending and tracking domains are successfully added on the list page (either the old one or the new one)
1. Be sure to verify adding tracking and sending domains that are assigned to subaccounts

### To Do

- [X] Wrap up remaining Cypress tests for tracking domains flow
- [x] Get questions answered by UX - **meeting set up on 9/23**
  - [x] Incorporate link to docs: https://www.sparkpost.com/docs/getting-started/requirements-for-sending-domains/
